toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/utilityassociationselement/UtilityAssociations.kt

```diff
@@ -0,0 +1,329 @@
+ * @param modifier The [Modifier] to apply to this layout.
+ */
+@Composable
+internal fun UtilityAssociationFilter(
+    groupResults: List<UtilityAssociationGroupResult>,
+    onGroupClick: (UtilityAssociationGroupResult) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    // show the list of layers
+    Surface(
+        modifier = modifier,
+        shape = RoundedCornerShape(15.dp)
+    ) {
+        LazyColumn(modifier = Modifier) {
+            groupResults.forEachIndexed { index, group ->
+                item {
+                    ListItem(
+                        headlineContent = {
+                            Text(text = group.name, modifier = Modifier.padding(start = 16.dp))
+                        },
+                        trailingContent = {
+                            Text(
+                                text = "${group.associationResults.count()}",
+                                modifier = Modifier.padding(end = 16.dp)
+                            )
+                        },
+                        modifier = Modifier.clickable {
+                            onGroupClick(group)
+                        },
+                        colors = ListItemDefaults.colors(
+                            containerColor = MaterialTheme.colorScheme.surfaceContainer
+                        )
+                    )
+                    if (index < groupResults.count() - 1) {
+                        Surface(
+                            modifier = Modifier.fillMaxWidth(),
+                            color = MaterialTheme.colorScheme.surfaceContainer
+                        ) {
+                            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Displays the provided list of associations that are part of the [UtilityAssociationGroupResult].
+ *
+ * @param groupResult The [UtilityAssociationGroupResult] to display.
+ * @param onItemClick A callback that is called when an association is clicked.
+ * @param onDetailsClick A callback that is called when the details icon is clicked.
+ * @param modifier The [Modifier] to apply to this layout.
+ */
+@Composable
+internal fun UtilityAssociations(
+    groupResult: UtilityAssociationGroupResult,
+    isNavigationEnabled: Boolean,
+    onItemClick: (Int) -> Unit,
+    onDetailsClick: (Int) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val lazyListState = rememberLazyListState()
+    Surface(
+        modifier = modifier.wrapContentHeight(
+            align = Alignment.Top
+        ),
+        shape = RoundedCornerShape(15.dp),
+        color = MaterialTheme.colorScheme.surfaceContainer
+    ) {
+        LazyColumn(
+            modifier = Modifier.clip(shape = RoundedCornerShape(15.dp)),
+            state = lazyListState
+        ) {
+            groupResult.associationResults.forEachIndexed { index, info ->
```

Comment:
```suggestion
            groupResult.associationResults.forEachIndexed { index, info ->
```
IIRC popup spec defines a [displayCount](https://runtime-zip.esri.com/userContent/runtime/kotlin/doc_kotlin/300.0.0/last_good/html/arcgis-maps-kotlin/com.arcgismaps.mapping.popup/-utility-associations-popup-element/display-count.html) for the number of associations to show. I would check with Swift to see how they've handles this.

Reply:
I have added support for taking in account the display count.